File: skills/java-mysql-query/scripts/CicdHelper.java
package scripts;
import java.io.*;
import java.nio.file.*;

/**
 * CI/CD 集成助手 (Java 版)
 * 自动化审计：校验Git提交信息，运行数据质量检查，生成审计报告。
 * 用法: javac -encoding utf8 scripts/CicdHelper.java
 *       java -cp . scripts.CicdHelper --check-commit-msg "feat(user): 新增接口"
 */
public class CicdHelper {
    public static void main(String[] args) throws Exception {
        String command = "";
        String commitMsg = "", auditFile = "", outputDir = "./reports";
        for (int i = 0; i < args.length; i++) {
            switch (args[i]) {
                case "--check-commit-msg": commitMsg = args[++i]; command = "check-commit"; break;
                case "--pre-commit-install": command = "install-hook"; break;
                case "--audit": auditFile = args[++i]; command = "audit"; break;
                case "--output-dir": outputDir = args[++i]; break;
            }
        }
        switch (command) {
            case "check-commit":
                StringBuilder msgs = new StringBuilder("[");
                if (commitMsg.isEmpty()) msgs.append("\"提交信息为空\"");
                else if (!commitMsg.matches("^(feat|fix|refactor|test|docs|chore|perf|style|ci)\\([\\w-]+\\)?:\\s.+"))
                    msgs.append("\"格式不符: <类型>(<范围>): <描述>\"");
                else msgs.append("\"提交信息格式正确\"");
                msgs.append("]");
                String status = msgs.toString().contains("正确") ? "ok" : "error";
                System.out.println("{\"status\":\"" + status + "\",\"messages\":" + msgs + "}");
                break;
            case "install-hook":
                Paths.get(".git/hooks").toFile().mkdirs();
                String hook = "#!/bin/sh\njava -cp . scripts.CicdHelper --check-commit-msg \"$(git log -1 --pretty=%B)\"\n";
                Files.write(Paths.get(".git/hooks/pre-commit"), hook.getBytes());
                new File(".git/hooks/pre-commit").setExecutable(true);
                System.out.println("{\"status\":\"installed\"}");
                break;
            case "audit":
                Files.createDirectories(Paths.get(outputDir));
                ProcessBuilder pb = new ProcessBuilder("java", "-cp", ".", "scripts.AuditReportGenerator",
                    "--input", auditFile, "--format", "markdown", "--output", outputDir + "/audit_report.md");
                pb.inheritIO().start().waitFor();
                System.out.println("{\"status\":\"success\",\"output\":\"" + outputDir + "/audit_report.md\"}");
                break;
            default:
                System.out.println("用法: --check-commit-msg <msg> | --pre-commit-install | --audit <file> --output-dir <dir>");
        }
    }
}

File: skills/java-mysql-query/scripts/cicd-helper.js
#!/usr/bin/env node
/**
 * CI/CD 集成助手 (Node.js 版)
 * 自动化审计：校验Git提交信息，运行数据质量检查，生成审计报告。
 * 用法: node cicd-helper.js --check-commit-msg "feat(user): 新增接口"
 *       node cicd-helper.js --pre-commit-install
 */
const fs = require("fs");
const { MySQLQuery } = require("./database-query.js");
const path = require("path");

function checkCommitMessage(msg) {
  const errors = [];
  const pattern = /^(feat|fix|refactor|test|docs|chore|perf|style|ci)(\([\w-]+\))?: .+/;
  if (!msg) errors.push("提交信息为空");
  else if (!pattern.test(msg)) errors.push("格式不符: <类型>(<范围>): <描述>");
  if (msg.length > 100) errors.push("提交信息超过100字符");
  return errors.length > 0 ? errors : ["提交信息格式正确"];
}

function parseArgs() {
  const args = process.argv.slice(2);
  const opts = {};
  for (let i = 0; i < args.length; i++) {
    switch (args[i]) {
      case "--check-commit-msg": opts.checkCommitMsg = args[++i]; break;
      case "--pre-commit-install": opts.preCommitInstall = true; break;
      case "--audit": opts.audit = args[++i]; break;
      case "--output-dir": opts.outputDir = args[++i]; break;
    }
  }
  return opts;
}

function run(opts) {
  if (opts.checkCommitMsg) {
    const result = checkCommitMessage(opts.checkCommitMsg);
    console.log(JSON.stringify({ status: result[0].includes("正确") ? "ok" : "error", messages: result }));
  } else if (opts.preCommitInstall) {
    const hookDir = path.join(".git", "hooks");
    if (!fs.existsSync(hookDir)) fs.mkdirSync(hookDir, { recursive: true });
    const hookPath = path.join(hookDir, "pre-commit");
    fs.writeFileSync(hookPath, '#!/bin/sh\nnode scripts/cicd-helper.js --check-commit-msg "$(git log -1 --pretty=%B)"\n', { mode: 0o755 });
    console.log(JSON.stringify({ status: "installed", hook: hookPath }));
  } else if (opts.audit) {
    const outDir = opts.outputDir || "./reports";
    if (!fs.existsSync(outDir)) fs.mkdirSync(outDir, { recursive: true });
    const outPath = path.join(outDir, "audit_report.md");
    execSync(`node scripts/audit-report-generator.js --input "${opts.audit}" --format markdown --output "${outPath}"`, { shell: true });
    console.log(JSON.stringify({ status: "success", output: outPath }));
  } else {
    console.log("用法: --check-commit-msg <msg> | --pre-commit-install | --audit <file> --output-dir <dir>");
  }
}
run(parseArgs());
